bindings/java/java/src/main/java/org/ethereum/evmc/EvmcVm.java

```diff
@@ -0,0 +1,113 @@
+package org.ethereum.evmc;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The Java interface to the evm instance.
+ *
+ * <p>Defines the Java methods capable of accessing the evm implementation.
+ */
+public final class EvmcVm {
```

Comment:
Are you sure those methods need to be static?

Reply:
This is a good question.  I've gone back and forth.  One thing to consider when making the decision is the fact that i need to maintain a single contextList to track the HostContexts.  If i changed EvmcVm methods to be non-static then i would have to offload that responsibility to the user of the api.  Open to suggestions here.  @mkalinin @atoulme, thoughts?

Reply:
This depends on the lifecycle and would be a good question for the EVMC maintainers. Do you create a VM once for the duration of the whole life of the client? Or do you create one every time? From my experience, EVMs are very shortlived. They have some state in them - they maintain a memory. So I've seen EVMs being created for each transaction execution, so they're short lived and can garbage collected quickly. The context object itself tends to contain the changeset of the execution, and are then applied to the world state.

Reply:
It's the other way around. You can create single EVMC VM for the client lifespan. The execute method can be invoked multiple times and concurrently - https://evmc.ethereum.org/group__EVMC.html#gaed9a4ab5609b55c5e3272d6d37d84ff7 (I might need to mention concurrently there). The execute method will create the needed execute context and dispose it when execution is finished. You can think about the VM "create" function as a way to init an external module.

Reply:
done